### ISSUE TITLE
Building dagger fails when alpinelinux.org is down

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#9128]
## Overview

When alpinelinux.org is down, building dagger fails. This may also affect other dagger modules that use alpine in the same way we do.

## What I did

```
dagger call -m github.com/dagger/dagger dev
```

## What I expected

A successful build of the latest dagger engine

## What happened

*Intermittently* I get:

```
[...]
│ ✔ alpine(branch: "3.20.2", packages: ["build-base", "go", "git"]): Alpine! 0.2s
│ ∅ .container: Container! 0.2s
│ ! failed to get alpine releases: failed to get alpine releases: Get "https://alpinelinux.org/releases.json": dial tcp [2a01:7e00:e000:2fc::4]:443: connect: network is unreachable
│
│ ✔ alpine(branch: "3.20.2"): Alpine! 0.1s
│ ✘ .container: Container! 0.2s
│ ! failed to get alpine releases: failed to get alpine releases: Get "https://alpinelinux.org/releases.json": dial tcp [2a01:7e00:e000:2fc::4]:443: connect: network is unreachable
```

Trace: https://dagger.cloud/dagger/traces/747a73358ef6234caaa6c8042ff3cc16

**Comment by jedevc** [comment]
Agreed, this is a little frustrating - however, this is the case for a whole variety of services:
- Dockerhub (for our use of official images)
- The go proxy, pythonhosted, npm (anything that our SDKs use)
- Github (or anywhere our modules are hosted)

There's a few things we can do to improve this generally, but having better caching is really the best first step. Avoiding making as many network requests as possible will be a huge help here.

After that, I think having a concept of "pinned" values is what we need to remove those last few network requests - e.g. we *always* request https://alpinelinux.org/releases.json on every run right now - but if a user pins this to a specific value (using a feature like https://github.com/dagger/dagger/issues/4679), then we could avoid even this.

Sadly, on a run with no cache available, there's not really a good solution here - if we can't fetch network deps for a build, we can't do the build.
